File: packages/typia/src/transformers/features/misc/MiscCreatePruneTransformer.ts
import { MiscPruneProgrammer } from "../../../programmers/misc/MiscPruneProgrammer";

import { ITransformProps } from "../../ITransformProps";
import { GenericTransformer } from "../../internal/GenericTransformer";

export namespace MiscCreatePruneTransformer {
  export const transform = (props: ITransformProps) =>
    GenericTransformer.factory({
      ...props,
      method: "misc.createPrune",
      write: MiscPruneProgrammer.write,
    });
}

File: packages/typia/native/transform/features/misc/MiscCreatePruneTransformer.go
package misc

import (
	shimast "github.com/microsoft/typescript-go/shim/ast"
	nativecontext "github.com/samchon/typia/packages/typia/native/core/context"
	nativemiscprogrammers "github.com/samchon/typia/packages/typia/native/core/programmers/misc"
	nativeinternal "github.com/samchon/typia/packages/typia/native/transform/internal"
)

type miscCreatePruneTransformerNamespace struct{}

var MiscCreatePruneTransformer = miscCreatePruneTransformerNamespace{}

func (miscCreatePruneTransformerNamespace) Transform(props nativeinternal.ITransformProps) *shimast.Node {
	return nativeinternal.GenericTransformer.Factory(nativeinternal.GenericTransformer_IProps{
		ITransformProps: props,
		Method:          "misc.createPrune",
		Write: func(x nativecontext.IProgrammerProps) *shimast.Node {
			return nativemiscprogrammers.MiscPruneProgrammer.Write(x)
		},
	})
}
